homeassistant/components/enphase_envoy/__init__.py

```diff
@@ -45,14 +47,22 @@ async def async_setup_entry(hass: HomeAssistant, entry: EnphaseConfigEntry) -> b
         hass.config_entries.async_update_entry(entry, unique_id=envoy.serial_number)
 
     if entry.unique_id != envoy.serial_number:
+        if TYPE_CHECKING:
+            assert entry.unique_id is not None
+            assert envoy.serial_number is not None
```

Comment:
Shouldn't these be casted instead of asserted?

Reply:
That is an option as well. I've had both recommendations over time.